pywt/_extensions/_swt.pyx

```diff
@@ -184,15 +178,18 @@ cpdef swt_axis(np.ndarray data, Wavelet wavelet, size_t level,
                     common.COEF_DETAIL, common.MODE_PERIODIZATION,
                     i, common.SWT_TRANSFORM)
             if retval:
-                raise RuntimeError("C wavelet transform failed")
+                raise RuntimeError(
+                    "C wavelet transform failed with error code %d" % retval)
         else:
             raise TypeError("Array must be floating point, not {}"
                             .format(data.dtype))
         ret.append((cA, cD))
 
         # previous approx coeffs are the data for the next level
         data = cA
-        data_info = output_info
+        # update data_info to match the new data array
+        data_info.strides = <pywt_index_t *> data.strides
+        data_info.shape = <size_t *> data.shape
```

Comment:
``data_info`` is overwritten at the top of this for-loop, so these lines do nothing.